src/ImageDetector/index.js

```diff
@@ -0,0 +1,162 @@
+import * as tf from '@tensorflow/tfjs';
+import handleArguments from '../utils/handleArguments';
+import { mediaReady } from '../utils/imageUtilities';
+
+/**
+ * @typedef {Object} SpecificDetectorImplementation
+ *
+ * @property {Promise<boolean>} ready - lets the parent detector know that the
+ * specific implementation is ready.
+ *
+ * @property {(input: tf.Tensor3D) => Promise<Array>} detect - core detection method.
+ * the ImageDetector will call the `detect` method of the specific implementation.
+ * It should accept a TensorFlow tensor?? or an image??
+ * And return an array of detections??
+ */
+
+/**
+ * Helper class for handling the public API of detector models (facemesh, etc.)
+ * Exposes the public methods for single and continuous detection.
+ * Executes the detection using whatever model is passed to the constructor.
+ */
+export default class ImageDetector {
+
+  /**
+   * @param {SpecificDetectorImplementation} specificImplementation
+   */
+  constructor(specificImplementation) {
+    /**
+     * @type {SpecificDetectorImplementation}
+     */
+    this.implementation = specificImplementation;
+
+    /**
+     * @type {Promise<boolean>}
+     * TODO: do we need to handle onReady callbacks?
+     */
+    this.ready = this.implementation.ready;
```

Comment:
I'm thinking I should handle the `callCallback` stuff here instead of in each model.